sshconnect2.c

```diff
@@ -1266,10 +1266,22 @@ identity_sign(struct identity *id, u_char **sigp, size_t *lenp,
 			goto out;
 		}
 		sign_key = prv;
-		if (sshkey_is_sk(sign_key)) {
+		if (sshkey_is_sk(sign_key) &&
+		    (sign_key->sk_flags & SSH_SK_USER_PRESENCE_REQD)) {
+			notifier = notify_start(options.batch_mode,
+			    "You may need to touch your authenticator.");
+			notify_complete(notifier, NULL);
+			notifier = NULL;
```

Comment:
This section will behave badly with SSH_ASKPASS - the dialog will executed and then be killed instantly. The `notify_complete()` call has to happen after `sshkey_sign()` completes